tests/cupyx_tests/scipy_tests/interpolate_tests/test_polyint.py

```diff
@@ -0,0 +1,256 @@
+    def test_shapes_vectorvalue_3(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind == 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([4, -5, 3, 2, -4])
+        xs = xp.linspace(-10, 10, 20, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(
+            xs,
+            xp.outer(ys, xp.arange(3, dtype=dtype)))
+        return P([0, 1])
+
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-4, rtol=1e-4)
+    def test_shapes_vectorvalue_3_float16(self, xp, scp):
+        true_poly = xp.poly1d([4, -5, 3, 2, -4])
+        xs = xp.linspace(-1, 1, 20, dtype=xp.float16)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(
+            xs,
+            xp.outer(ys, xp.arange(3, dtype=xp.float16)))
+        return P([0, 1])
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_shapes_1d_vectorvalue_1(self, xp, scp, dtype):
+        true_poly = xp.poly1d([-3, -1, 4, 9, 8])
+        xs = xp.linspace(-1, 10, 10, dtype=dtype)
+        ys = true_poly(xs)
+        return xp.shape(scp.interpolate.BarycentricInterpolator(
+                        xs,
+                        xp.outer(ys, xp.array([1], dtype=dtype)))(0))
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_shapes_1d_vectorvalue_2(self, xp, scp, dtype):
+        true_poly = xp.poly1d([-3, -1, 4, 9, 8])
+        xs = xp.linspace(-1, 10, 10, dtype=dtype)
+        ys = true_poly(xs)
+        return xp.shape(scp.interpolate.BarycentricInterpolator(
+                        xs,
+                        xp.outer(ys, xp.array([1], dtype=dtype)))([0]))
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_shapes_1d_vectorvalue_3(self, xp, scp, dtype):
+        true_poly = xp.poly1d([-3, -1, 4, 9, 8])
+        xs = xp.linspace(-1, 1, 10, dtype)
+        ys = true_poly(xs)
+        return xp.shape(scp.interpolate.BarycentricInterpolator(
+                        xs,
+                        xp.outer(ys, xp.array([1], dtype=dtype)))([0, 1]))
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_large_chebyshev(self, xp, scp, dtype):
+        n = 100
+        j = xp.arange(n + 1, dtype=dtype).astype(xp.float64)
+        x = xp.cos(j * xp.pi / n)
+
+        w = (-1) ** j
+        w[0] *= 0.5
+        w[-1] *= 0.5
+
+        return scp.interpolate.BarycentricInterpolator(x).wi
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_complex_1(self, xp, scp, dtype):
+        x = xp.array([1, 2, 3, 4], dtype=dtype)
+        y = xp.array([1, 2, 1j, 3])
+        return scp.interpolate.BarycentricInterpolator(x, y)(x)
```

Comment:
I think it makes less sense to pass `x` to the interpolator as it is constructed to pass through the points given by `x` and `y`.

Reply:
Okay, created another array and used that as an input.